torch/onnx/_internal/fx/export.py

```diff
@@ -500,6 +559,184 @@ def _validate_op_between_ort_torch(
         warnings.warn(f"ORT fails to run with error: {e}.")
 
 
+def _location_from_fx_stack_trace(
+    node_stack_trace: str,
+) -> Optional[diagnostics.infra.Location]:
+    """Extract location from FX node stack trace.
+
+    Args:
+        node_stack_trace: The stack trace of the FX node. Example:
+
+            File "path/file.py", line 311, in <function>
+                <code>
+            |   File "path/file2.py", line 389, in <function>
+                <code>
+
+    Returns:
+        location: The location of the FX node.
+    """
+    lines = node_stack_trace.strip().split("\n")
+    idx = 0
+    while idx < len(lines) and "File" not in lines[idx]:
+        idx += 1
+    if idx + 1 >= len(lines):
+        return None
+
+    pattern = re.compile(r"^File \"(.+)\", line (\d+), in (.+)$")
+    matches = pattern.match(lines[idx].strip())
+    if matches:
+        uri = matches.group(1)
+        line_number = int(matches.group(2))
+        snippet = lines[idx + 1].strip()
+        return diagnostics.infra.Location(uri=uri, line=line_number, snippet=snippet)
+    return None
+
+
+@diagnostics.diagnose_call(
+    rule=diagnostics.rules.fx_node_to_onnx,
+    exception_report_level=diagnostics.levels.ERROR,
+)
+def _export_fx_node_to_onnxscript(
+    node: torch.fx.Node,
+    onnxscript_graph,
+    fx_name_to_onnxscipt_value,
+    onnxscript_value_name_to_real_tensor,
+    tracer,
+    fx_module_with_metadata,
+):
+    # Record stack trace of node in diagnostic.
+    node_stack_trace = node.stack_trace
+    if node_stack_trace:
```

Comment:
if otherwise it gives None.

```suggestion
    if node_stack_trace is not None:
```

to differentiate with boolean.

Reply:
`node.stack_trace` is type `Optional[str]`. The way the code is written is intended, to also leave out empty string.